core/trino-main/src/test/java/io/trino/security/TestFileBasedSystemAccessControl.java

```diff
@@ -632,6 +634,79 @@ public void testViewOperationsReadOnly()
+        AccessControlManager accessControlManager = newAccessControlManager(transactionManager, "catalog.json");
+
+        transaction(transactionManager, accessControlManager)
+                .execute(transactionId -> {
+                    SecurityContext aliceContext = new SecurityContext(transactionId, alice, queryId);
+                    SecurityContext bobContext = new SecurityContext(transactionId, bob, queryId);
+                    SecurityContext nonAsciiContext = new SecurityContext(transactionId, nonAsciiUser, queryId);
+
+                    accessControlManager.checkCanCreateView(aliceContext, aliceMaterializedView);
+                    accessControlManager.checkCanDropMaterializedView(aliceContext, aliceMaterializedView);
+
+                    accessControlManager.checkCanCreateView(aliceContext, staffMaterializedView);
+                    accessControlManager.checkCanDropMaterializedView(aliceContext, staffMaterializedView);
+
+                    assertThatThrownBy(() -> accessControlManager.checkCanCreateView(bobContext, aliceMaterializedView))
+                            .isInstanceOf(AccessDeniedException.class)
+                            .hasMessage("Access Denied: Cannot access catalog alice-catalog");
+                    assertThatThrownBy(() -> accessControlManager.checkCanDropMaterializedView(bobContext, aliceMaterializedView))
+                            .isInstanceOf(AccessDeniedException.class)
+                            .hasMessage("Access Denied: Cannot access catalog alice-catalog");
+
+                    accessControlManager.checkCanCreateView(bobContext, staffMaterializedView);
+                    accessControlManager.checkCanDropMaterializedView(bobContext, staffMaterializedView);
+
+                    assertThatThrownBy(() -> accessControlManager.checkCanCreateView(nonAsciiContext, aliceMaterializedView))
+                            .isInstanceOf(AccessDeniedException.class)
+                            .hasMessage("Access Denied: Cannot access catalog alice-catalog");
+                    assertThatThrownBy(() -> accessControlManager.checkCanDropMaterializedView(nonAsciiContext, aliceMaterializedView))
+                            .isInstanceOf(AccessDeniedException.class)
+                            .hasMessage("Access Denied: Cannot access catalog alice-catalog");
+
+                    assertThatThrownBy(() -> accessControlManager.checkCanCreateView(nonAsciiContext, staffMaterializedView))
+                            .isInstanceOf(AccessDeniedException.class)
+                            .hasMessage("Access Denied: Cannot access catalog staff-catalog");
+                    assertThatThrownBy(() -> accessControlManager.checkCanDropMaterializedView(nonAsciiContext, staffMaterializedView))
+                            .isInstanceOf(AccessDeniedException.class)
+                            .hasMessage("Access Denied: Cannot access catalog staff-catalog");
```

Comment:
I don't see explicit check for `materialized-view` name. Should there be one?

Reply:
Message is correct because the access control rule in test is defined at catalog level (see `catalog.json`)